### ISSUE TITLE
Bracket pair has problem coloring html and jsx

### ISSUE THREAD

**Issue by KROSF** [microsoft/vscode#130996]
@KROSF It's better to create separate issue as this is PR that already merged.
Looks like `/` is not colored and therefore whole ligature doesn't get color. Try to set `"editor.fontLigatures": false`

_Originally posted by @IllusionMH in https://github.com/microsoft/vscode/issues/129231#issuecomment-900300954_

**Comment by KROSF** [comment]
![image](https://user-images.githubusercontent.com/24454660/129732854-a2c2d5c4-096e-4e76-826e-69bc7f75d045.png)

There are some problems with closing jsx tags

![image](https://user-images.githubusercontent.com/24454660/129734434-00e6ab3f-2aa6-42f7-b88d-f7c5c1809b5a.png)
same problem with html

Also tested with `"editor.fontLigatures": false` as  @IllusionMH suggested

**Comment by hediet** [comment]
Thanks for reporting this bug!

Can you post the content of the document?

It does not happen for me (using Cascadia and editor.fontLigatures: true):
![image](https://user-images.githubusercontent.com/2931520/129737272-671cb103-40d2-40b7-a387-f666967be32f.png)

**Comment by KROSF** [comment]
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    
</body>
</html>
```

```jsx
const Loader = () => (
  <Container>
    <Panda />
    <Message>{MESSAGE}</Message>
  </Container>
);
```

**Comment by IllusionMH** [comment]
Try `<MyComponent />`
![image](https://user-images.githubusercontent.com/1312662/129738073-11edc62c-b77a-41e4-8825-7cc49d938c22.png)
![image](https://user-images.githubusercontent.com/1312662/129738127-8994bffd-cb38-496b-958d-3ab24b3545db.png)

`</` looks good in ligature if `<` is colored as expected.

**Comment by hediet** [comment]
> ```
> <!DOCTYPE html>
> <html lang="en">
> <head>
>     <meta charset="UTF-8">
>     <meta http-equiv="X-UA-Compatible" content="IE=edge">
>     <meta name="viewport" content="width=device-width, initial-scale=1.0">
>     <title>Document</title>
> </head>
> <body>
>     
> </body>
> </html>
> ```

This is how it looks for me:

![image](https://user-images.githubusercontent.com/2931520/129739219-d96fba36-7dd6-4d56-a493-936cdcf098f8.png)

**Comment by hediet** [comment]
> Try `<MyComponent />`
> ![image](https://user-images.githubusercontent.com/1312662/129738073-11edc62c-b77a-41e4-8825-7cc49d938c22.png)
> ![image](https://user-images.githubusercontent.com/1312662/129738127-8994bffd-cb38-496b-958d-3ab24b3545db.png)
> 
> `</` looks good in ligature if `<` is colored as expected.

How can I reproduce that?

**Comment by IllusionMH** [comment]
That's whole code I use in Untitled file with TypeScript React language

```json
    "editor.fontFamily": "'Cascadia Code PL', Consolas, 'Courier New', 'Segoe UI emoji', monospace",
    // "editor.fontLigatures": true,
```

**Comment by KROSF** [comment]
> Thanks for reporting this bug!
> 
> Can you post the content of the document?
> Can you try disabling all extensions and see if it still happens?
> 
> It does not happen for me (using Cascadia and editor.fontLigatures: true):
> 
> ![image](https://user-images.githubusercontent.com/2931520/129737272-671cb103-40d2-40b7-a387-f666967be32f.png)
> 
> I doubt it is a problem with recognizing `<`, otherwise the closing bracket would also not be colored:
> 
> ![image](https://user-images.githubusercontent.com/2931520/129737744-1295b2f4-b454-42a9-b2d4-a2ec3a59951f.png)

tested with all extensions disabled and still happens `code-insiders . --disable-extensions`

with `fontLigatures: false`

![image](https://user-images.githubusercontent.com/24454660/129742543-5218a071-d8c7-4f21-a3ce-5bbfc9e6ae62.png)

with `fontLigatures: true`

![image](https://user-images.githubusercontent.com/24454660/129742605-036e93a2-d1cc-4826-9514-b936e9274924.png)

**Comment by hediet** [comment]
It does not seem to be possible to give parts of a ligature a different color, as [this fiddle](https://jsfiddle.net/xL4ms6e8/) demonstrates.

> with `fontLigatures: true`
> 
> ![image](https://user-images.githubusercontent.com/24454660/129742605-036e93a2-d1cc-4826-9514-b936e9274924.png)

This screenshot I don't understand. Why is `/>` yellow?

**Comment by KROSF** [comment]
@hediet that is the correct behaviour, isn't it?
the problem is with the first screenshot that is not using '/>' as a single token (at least when language is html or jsx) to apply the same color

**Comment by hediet** [comment]
> @hediet that is the correct behaviour, isn't it?

No, it should only highlight `<` and `>`. In case of enabled ligatures and `/>`, only `>` should have a color set (which is ignored by electron), so the outcome should be that `/>` is uncolored.

Probably ligatures should be disabled for text sequences of differently styled characters.

**Comment by IllusionMH** [comment]
What is your OS and what font you are using?
Is it reproducible with all extensions disabled? You can try this with <kbd>F1</kbd> and `>Developer: Reload Window With Extensions Disabled`

Because result in your screenshots differs from what I see (and I assume **hediet** as well)
`</` **doesn't have** coloring, but as `<` is colored then whole ligature should be colored
`/>` **has** coloring, but as `/` is uncolored - whole ligature shouldn't be colored

in assumption that ligature is colored by first character in sequence

**Comment by hediet** [comment]
Closing, as being fixed by either #131194 (workaround) or #131092.

**Comment by rzhao271** [comment]
@hediet is this issue verifiable? It seems like it is being tracked in another issue.
If the issue is not verifiable, it can just be moved off the milestone
